dynamodb-stream-reader/src/main/java/elasticsearch/ElasticSearchRestClient.java

```diff
@@ -0,0 +1,127 @@
+    public static final String DELETE_LOG_MESSAGE = "Deleting from search API publication with identifier: {}";
+    public static final String ELASTICSEARCH_ENDPOINT_INDEX_KEY = "ELASTICSEARCH_ENDPOINT_INDEX";
+    public static final String ELASTICSEARCH_ENDPOINT_ADDRESS_KEY = "ELASTICSEARCH_ENDPOINT_ADDRESS";
+    public static final String ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY = "ELASTICSEARCH_ENDPOINT_API_SCHEME";
+    public static final String ELASTICSEARCH_ENDPOINT_URI_TEMPLATE = "%s://%s/%s/%s/%s";
+    public static final String POSTING_TO_ENDPOINT_LOG_MESSAGE = "POSTing {} to endpoint {}";
+
+    private final HttpClient client;
+    private final String elasticSearchEndpointAddress;
+    private final String elasticSearchEndpointIndex;
+    private final String elasticSearchEndpointScheme;
+
+    /**
+     * Creates a new ElasticSearchRestClient.
+     *
+     * @param httpClient Client to speak http
+     * @param environment Environment with properties
+     */
+    public ElasticSearchRestClient(HttpClient httpClient, Environment environment) {
+        client = httpClient;
+        elasticSearchEndpointAddress = environment.readEnv(ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        elasticSearchEndpointIndex = environment.readEnv(ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        elasticSearchEndpointScheme = environment.readEnv(ELASTICSEARCH_ENDPOINT_API_SCHEME_KEY);
+        logger.info(INITIAL_LOG_MESSAGE,
+                elasticSearchEndpointScheme, elasticSearchEndpointAddress, elasticSearchEndpointIndex);
+    }
+
+    /**
+     * Adds or insert a document to an elasticsearch index.
+     * @param document the document to be inserted
+     * @throws URISyntaxException thrown when uri is misconfigured
+     * @throws IOException thrown hen service i not available
+     * @throws InterruptedException thrown when service i interrupted
+     */
+    public void addDocumentToIndex(Map<String, String> document)
+            throws URISyntaxException, IOException, InterruptedException {
+        logger.debug(UPSERTING_LOG_MESSAGE, document);
+
+        HttpRequest request = createHttpRequest(document);
+
+        HttpResponse<String> response = doSend(request);
+        logger.debug(response.body());
+    }
+
+    private HttpRequest createHttpRequest(Map<String, String> document) throws
+            JsonProcessingException, URISyntaxException {
+        String requestBody = JsonUtils.objectMapper.writeValueAsString(document);
+        String identifier = document.get(IDENTIFIER_KEY);
+
+        HttpRequest request = buildHttpRequest(requestBody, identifier);
+
+        logger.debug(POSTING_TO_ENDPOINT_LOG_MESSAGE, requestBody, createUpsertDocumentURI(identifier));
+        return request;
+    }
+
+    private HttpRequest buildHttpRequest(String requestBody, String identifier) throws URISyntaxException {
+        HttpRequest request = HttpRequest.newBuilder()
```

Comment:
remove variable and return inline variable

Reply:
Oh yes...